lang/src/test/kotlin/org/partiql/lang/eval/NaturalExprValueComparatorsTest.kt

```diff
@@ -363,13 +364,13 @@ class NaturalExprValueComparatorsTest : EvaluatorTestBase() {
         }
         val orderedIter = ordered.iterator()
         case.expected.map { it.toMutableList() }.forEach { equivs ->
```

Comment:
What is the reasoning for updating this test in the context of this PR?

Reply:
This is because factory pattern has memoized states for constructing boolean `ExprValue`, but not for `ExprValue` transformed from `IonValue`. With the new API, either directly constructing boolean `ExprValue` or get it from `IonValue`, we are using the memoized states. So referential equivalence does not work this time for this test suite.